src/common/Finisher.cc

```diff
@@ -3,6 +3,11 @@
 
 #include "Finisher.h"
 
+#include "common/debug.h"
+
+// re-include our assert to clobber the system one; fix dout:
```

Comment:
@chardan i admit this is mess and we need to fix in long term without the `_ASSERT_H` hackery .

but i am still just wondering why we need to add this in this change. but don't need before. is there any headers we are introducing break it?

also, you need to avoid the fix-of-a-fix commits in a single PR.